### PR TITLE
Add the `keys.typedmethodkey` decorator

### PR DESCRIPTION
**Pull request by kuraga**
Closes: https://github.com/tkem/cachetools/issues/296

### CODE REVIEW

**Comment by tkem** [comment]
Sorry for the delay, but I promise I'll look into this in more detail by the end of April, latest!

File: docs/index.rst [hunk 1]
```diff
@@ -491,7 +491,7 @@ functions with the :func:`cached` and :func:`cachedmethod` decorators:
 
 .. autofunction:: methodkey
 
-   This function is equivalent to :func:`hashkey`, but ignores its
+   This function is similar to :func:`hashkey`, but ignores its
```
Comment (tkem) [inline docs/index.rst:494]: :+1:

File: tests/test_cachedmethod.py
Comment (tkem) [inline tests/test_cachedmethod.py:1]: Reading the tests alone, this makes much more sense than previously. So :+1:, thanks for your `commit`ment (sorry for the bad joke, couldn't resist)!

**Comment by tkem** [comment]
Merged into https://github.com/tkem/cachetools/tree/dev/v5.4.0, thanks!

**Comment by kuraga** [comment]
Thanks!

**Comment by tkem** [comment]
Should all be merged into v5.4.0.

**Comment by tkem** [comment]
No, thank _you_, ans sorry for the delay!

**Pull request closed without merging**